packages/space-opera/src/components/animation_controls/animation_controls.ts

```diff
@@ -42,10 +37,10 @@ interface AnimationControlsInterface {
 export class AnimationControls extends ConnectedLitElement {
   @query('me-checkbox#animation-autoplay') autoplayCheckbox?: CheckboxElement;
   @internalProperty() animationNames: string[] = [];
-  @internalProperty() config: AnimationControlsInterface = {};
+  @internalProperty() config: ModelViewerConfig = {};
 
   stateChanged(state: State) {
-    this.animationNames = getAnimationNames(state);
+    this.animationNames = getModelViewer()?.availableAnimations ?? [];
     this.config = getConfig(state);
```

Comment:
This may be leftover, but if the only thing this.config is used for in this component is animationName, you'd probably want to have a getAnimationName(state) method that returns this.config.animationName

Using getConfig() will cause you to re-render this component every time state.config changes, whereas you only want this component to re-render every time this.config.animationName changes (or any other values for this component)

Reply:
Ah, that's a very helpful insight! Will do.

Reply:
Fixed.